tests/test_window.py

```diff
@@ -310,3 +312,45 @@ def test_empty_window_name(session: Session) -> None:
         "#{==:#{session_name}," + session.name + "}",
     )
     assert "''" in cmd.stdout
+
+
+@pytest.mark.skipif(
+    has_lt_version("3.0"),
+    reason="needs -e flag for split-window which was introduced in 3.0",
+)
+def test_split_window_with_environment(session: Session) -> None:
+    env = shutil.which("env")
+    assert env is not None, "Cannot find usable `env` in Path."
+
+    window = session.new_window(window_name="split_window_with_environment")
+    pane = window.split_window(
+        shell=f"{env} PS1='$ ' sh",
+        environment={"ENV_VAR": "pane"},
+    )
+    assert pane is not None
+    # wait a bit for the prompt to be ready as the test gets flaky otherwise
+    time.sleep(0.05)
+    pane.send_keys("echo $ENV_VAR")
+    assert pane.capture_pane() == ["$ echo $ENV_VAR", "pane", "$"]
+
+
+@pytest.mark.skipif(
+    has_gte_version("3.0"),
+    reason="3.0 has the -e flag on split-window",
+)
+def test_split_window_with_environment_logs_warning_for_old_tmux(
+    session: Session,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    env = shutil.which("env")
+    assert env is not None, "Cannot find usable `env` in Path."
+
+    window = session.new_window(window_name="split_window_with_environment")
+    window.split_window(
+        shell=f"{env} PS1='$ ' sh",
+        environment={"ENV_VAR": "pane"},
```

Comment:
Are you familiar with [pytest.mark.parametrize](https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions)? Any objection to adding some examples with more variations of env variables?

Is this going to assert the variable is set in some way?

Reply:
I'm not familiar with this one, but can take a look (probably tomorrow earliest). Any specific idea of what kind of different environment variables might be beneficial to test (in this test-case it won't make a difference as the test checks whether the correct warning is shown to the user when an old version of tmux is used)?

> Is this going to assert the variable is set in some way?

For this test it's not asserted as it won't end up anywhere. It's just to use the same setup as the test case which actually sets up the environment and for that test the assertion happens in line 334.

Reply:
> For this test it's not asserted as it won't end up anywhere

Good point.

This whole conversation was meant to be in `test_split_window_with_environment` - where for some reason I didn't see the `assert. My mistake.

Reply:
Moved the rest of the conversation to the appropriate test https://github.com/tmux-python/libtmux/pull/453/files#r1014695165